File: app/src/main/java/tk/lorddarthart/videostreamplayer/PortraitFragment.java
package tk.lorddarthart.videostreamplayer;


import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.media.AudioManager;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.support.constraint.ConstraintLayout;
import android.support.constraint.ConstraintSet;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.URLUtil;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.VideoView;

import java.util.Timer;
import java.util.TimerTask;

public class PortraitFragment extends Fragment {

    private String ARG_PARAM1 = "param1";
    private String ARG_PARAM2 = "param2";
    private String param1;
    private String param2;
    private Uri mData;
    private View mView;
    private FullscreenActivity mActivity;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mActivity = (FullscreenActivity) context;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            param1 = getArguments().getString(ARG_PARAM1);
            param2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mView = inflater.inflate(R.layout.fragment_portrait, container, false);
        mView.findViewById(R.id.btnPlay).setTag(R.drawable.ic_baseline_play_arrow);
        mView.findViewById(R.id.videoView).setTag("first");
        mData = mActivity.getIntent().getData();
        if (mData == null) {
            final AlertDialog.Builder builder = new AlertDialog.Builder(mActivity)
                    .setTitle("Please enter URL")
                    .setCancelable(false);
            final EditText input = new EditText(mActivity);
            builder.setView(input);

            builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    try {
                        mData = Uri.parse(input.getText().toString());
                        if (!URLUtil.isValidUrl(mData.toString())) {
                            throw new Exception();
                        } else {
                            dialog.cancel();
                            mView.findViewById(R.id.btnPlay).performClick();
                        }
                    } catch (Exception e) {
                        Snackbar.make(mActivity.findViewById(android.R.id.content), mActivity.getResources().getString(R.string.errorurl), Snackbar.LENGTH_SHORT).show();
                        mActivity.recreate();
                    }
                }
            });
            builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            dialog.cancel();
                        }
                    }
            ).show();
        }
        ;
        // Кнопка проигрывания
        mView.findViewById(R.id.btnPlay).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mView.findViewById(R.id.btnPlay).getTag().equals(R.drawable.ic_baseline_play_arrow)) {
                    if (mView.findViewById(R.id.videoView).getTag().equals("first")) {
                        clickBtnPlay(mView, mData); // Если видео только открыли
                    } else if (mView.findViewById(R.id.videoView).getTag().equals("pause")) {
                        clickBtnContinue(mView); // После паузы
                    }
                } else if (mView.findViewById(R.id.btnPlay).getTag().equals(R.drawable.ic_baseline_pause)) {
                    clickBtnPause(mView);
                } else if (mView.findViewById(R.id.btnPlay).getTag().equals(R.drawable.ic_baseline_repeat)) {
                    clickBtnRepeat(mView);
                }
            }
        });
        mView.findViewById(R.id.videoView).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                hideNavBtns(mView);
            }
        });

        mView.findViewById(R.id.btnStop).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnStop(mView);
            }
        });

        mView.findViewById(R.id.btnForward).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnForward(mView);
            }
        });

        mView.findViewById(R.id.btnBackward).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnBackward(mView);
            }
        });

        mView.findViewById(R.id.btnFaster).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnFaster(mView);
            }
        });

        mView.findViewById(R.id.btnSlower).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnSlower(mView);
            }
        });

        mView.findViewById(R.id.btnVolumeOnOff).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickBtnMute(mView);
            }
        });
        return mView;
    }

    private void clickBtnMute(View view) {
        AudioManager am = (AudioManager) mActivity.getSystemService(Context.AUDIO_SERVICE);
        int volumeLevel = am.getStreamVolume(AudioManager.STREAM_MUSIC);
        if (volumeLevel != 0) {
            am.setStreamMute(AudioManager.STREAM_MUSIC, true);
            ((ImageView)view.findViewById(R.id.btnVolumeOnOff)).setImageResource(R.drawable.ic_baseline_volume_off);
            view.findViewById(R.id.btnVolumeOnOff).setTag(R.drawable.ic_baseline_volume_off);
        } else {
            am.setStreamMute(AudioManager.STREAM_MUSIC, false);
            ((ImageView)view.findViewById(R.id.btnVolumeOnOff)).setImageResource(R.drawable.ic_baseline_volume_up);
            view.findViewById(R.id.btnVolumeOnOff).setTag(R.drawable.ic_baseline_volume_up);
        }
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnForward(View view) {
        ((VideoView)view.findViewById(R.id.videoView)).seekTo(((VideoView)view.findViewById(R.id.videoView)).getDuration());
        ((VideoView)view.findViewById(R.id.videoView)).pause();
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_repeat);
        view.findViewById(R.id.btnPlay).setTag(R.drawable.ic_baseline_repeat);
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnBackward(View view) {
        ((VideoView)view.findViewById(R.id.videoView)).pause();
        ((VideoView)view.findViewById(R.id.videoView)).seekTo(0);
        ((VideoView)view.findViewById(R.id.videoView)).setTag("pause");
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_play_arrow);
        view.findViewById(R.id.btnPlay).setTag(R.drawable.ic_baseline_play_arrow);
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnFaster(View view) {
        ((VideoView)view.findViewById(R.id.videoView)).seekTo(((VideoView)view.findViewById(R.id.videoView)).getCurrentPosition() + 10000);
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnSlower(View view) {
        ((VideoView)view.findViewById(R.id.videoView)).seekTo(((VideoView)view.findViewById(R.id.videoView)).getCurrentPosition() - 10000);
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnStop(View view) {
        ((VideoView)view.findViewById(R.id.videoView)).pause();
        ((VideoView)view.findViewById(R.id.videoView)).seekTo(0);
        ((VideoView)view.findViewById(R.id.videoView)).setTag("first");
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_play_arrow);
        view.findViewById(R.id.btnPlay).setTag(R.drawable.ic_baseline_play_arrow);
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void hideNavBtns(View view) { // Скрыть-показать панель
        if (view.findViewById(R.id.clNavBtns).getVisibility() == View.VISIBLE) {
            view.findViewById(R.id.clNavBtns).setVisibility(View.GONE);
            new Timer().schedule(new MyTimerTask(), 5000);
        } else if (view.findViewById(R.id.clNavBtns).getVisibility() == View.GONE) {
            view.findViewById(R.id.clNavBtns).setVisibility(View.VISIBLE);
            new Timer().schedule(new MyTimerTask(), 5000);
        }
    }

    private void clickBtnContinue(View view) { // Продолжить после паузы
        ((VideoView)view.findViewById(R.id.videoView)).start();
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_pause);
        ((ImageView)view.findViewById(R.id.btnPlay)).setTag(R.drawable.ic_baseline_pause);
        view.findViewById(R.id.videoView).setTag("play");
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnPause(View view) { // Поставить паузу
        ((VideoView)view.findViewById(R.id.videoView)).pause();
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_play_arrow);
        ((ImageView)view.findViewById(R.id.btnPlay)).setTag(R.drawable.ic_baseline_play_arrow);
        view.findViewById(R.id.videoView).setTag("pause");
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnRepeat(View view) { // Начать заново
        ((VideoView)view.findViewById(R.id.videoView)).resume();
        ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_pause);
        ((ImageView)view.findViewById(R.id.btnPlay)).setTag(R.drawable.ic_baseline_pause);
        view.findViewById(R.id.videoView).setTag("play");
        new Timer().schedule(new MyTimerTask(), 5000);
    }

    private void clickBtnPlay(final View view, Uri uri) { // Запустить проигрывание
        // Показываем окно загрузки
        final ProgressDialog mDialog = new ProgressDialog(mActivity);

        try {
            mDialog.setMessage("Пожалуйста, подождите...");
            mDialog.setCanceledOnTouchOutside(false);
            mDialog.show();
            ((VideoView)view.findViewById(R.id.videoView)).setVideoURI(uri);
            ((VideoView)view.findViewById(R.id.videoView)).setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
                @Override
                public void onCompletion(MediaPlayer mp) {
                    ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_repeat);
                    ((ImageView)view.findViewById(R.id.btnPlay)).setTag(R.drawable.ic_baseline_repeat);
                    if (view.findViewById(R.id.clNavBtns).getVisibility() == View.GONE) {
                        view.findViewById(R.id.clNavBtns).setVisibility(View.VISIBLE);
                    }
                }
            });
        } catch (Exception e) {
            Snackbar.make(view, mActivity.getResources().getString(R.string.error), Snackbar.LENGTH_LONG)
                    .show(); // В случае ошибки показать Snackbar с текстом
        }
        ((VideoView)view.findViewById(R.id.videoView)).requestFocus();
        // Проигрыватель готов к воспроизведению
        ((VideoView)view.findViewById(R.id.videoView)).setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
            @Override
            public void onPrepared(MediaPlayer mediaPlayer) {
                mDialog.dismiss();
                mediaPlayer.setLooping(false);
                if (mediaPlayer.getVideoHeight() > mediaPlayer.getVideoWidth()) {
                    ((VideoView)view.findViewById(R.id.videoView)).setLayoutParams(new ConstraintLayout.LayoutParams(
                            ConstraintLayout.LayoutParams.WRAP_CONTENT,
                            ConstraintLayout.LayoutParams.MATCH_PARENT
                    ));
                    ConstraintSet constraintSet = new ConstraintSet();
                    constraintSet.clone((ConstraintLayout) view);
                    constraintSet.connect(((VideoView)view.findViewById(R.id.videoView)).getId(), ConstraintSet.LEFT, view.getId(), ConstraintSet.LEFT, 0);
                    constraintSet.connect(((VideoView)view.findViewById(R.id.videoView)).getId(), ConstraintSet.RIGHT, view.getId(), ConstraintSet.RIGHT, 0);
                    constraintSet.applyTo((ConstraintLayout) view);
                } else {
                    ((VideoView)view.findViewById(R.id.videoView)).setLayoutParams(new ConstraintLayout.LayoutParams(
                            ConstraintLayout.LayoutParams.MATCH_PARENT,
                            ConstraintLayout.LayoutParams.WRAP_CONTENT
                    ));
                    ConstraintSet constraintSet = new ConstraintSet();
                    constraintSet.clone((ConstraintLayout) view);
                    constraintSet.connect(((VideoView)view.findViewById(R.id.videoView)).getId(), ConstraintSet.TOP, view.getId(), ConstraintSet.TOP, 0);
                    constraintSet.connect(((VideoView)view.findViewById(R.id.videoView)).getId(), ConstraintSet.BOTTOM, view.getId(), ConstraintSet.BOTTOM, 0);
                    constraintSet.applyTo((ConstraintLayout) view);
                }
                ((VideoView)view.findViewById(R.id.videoView)).start();
                ((ImageView)view.findViewById(R.id.btnPlay)).setImageResource(R.drawable.ic_baseline_pause);
                ((ImageView)view.findViewById(R.id.btnPlay)).setTag(R.drawable.ic_baseline_pause);
                ((VideoView)view.findViewById(R.id.videoView)).setTag("play");
                new Timer().schedule(new MyTimerTask(), 5000);
            }
        });
    }

    private class MyTimerTask extends TimerTask {

        @Override
        public void run() {
            mActivity.runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    mView.findViewById(R.id.clNavBtns).setVisibility(View.GONE);
                }
            });
        }
    }
}

File: app/src/main/java/tk/lorddarthart/videostreamplayer/PortraitFragment.kt
@file:Suppress("DEPRECATION")

package tk.lorddarthart.videostreamplayer


import android.app.ProgressDialog
import android.net.Uri
import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_portrait.view.*
import java.lang.Exception
import android.support.constraint.ConstraintSet
import java.util.*
import android.app.AlertDialog
import android.content.Context
import android.media.AudioManager
import android.text.InputType
import android.webkit.URLUtil
import android.widget.EditText
import android.widget.Toast


// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 * Use the [PortraitFragment.newInstance] factory method to
 * create an instance of this fragment.
 *
 */
@Suppress("DEPRECATED_IDENTITY_EQUALS")
class PortraitFragment : Fragment() {
    // TODO: Rename and change types of parameters
    private var param1: String? = null
    private var param2: String? = null
    private var data: Uri? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_portrait, container, false)
        view.btnPlay.tag = R.drawable.ic_baseline_play_arrow
        view.videoView.tag = "first"
        data = activity?.intent?.data
        if (data == null) {
            val builder = AlertDialog.Builder(activity)
            builder.setTitle("Please enter URL")
            builder.setCancelable(false)
            val input = EditText(activity)
            builder.setView(input)

            builder.setPositiveButton(
                "OK"
            ) { dialog, _ ->
                try {
                    data = Uri.parse(input.text.toString())
                    if (!URLUtil.isValidUrl(data.toString())) {
                        throw Exception()
                    } else {
                        dialog.cancel()
                        view.btnPlay.performClick()
                    }
                } catch (e: Exception) {
                    Toast.makeText(context, resources.getString(R.string.errorurl), Toast.LENGTH_LONG).show()
                    activity!!.recreate()
                }
            }
            builder.setNegativeButton("Cancel") { dialog, _ -> dialog.cancel() }

            builder.show()
        }
        // Кнопка проигрывания
        view.btnPlay.setOnClickListener {
            when (view.btnPlay.tag) {
                R.drawable.ic_baseline_play_arrow -> {
                    when (view.videoView.tag) {
                        "first" -> clickBtnPlay(view, data!!) // Если видео только открыли
                        "pause" -> clickBtnContinue(view) // После паузы
                    }
                }
                R.drawable.ic_baseline_pause -> {
                    clickBtnPause(view)
                }
                R.drawable.ic_baseline_repeat -> {
                    clickBtnRepeat(view)
                }
            }
        }
        view.videoView.setOnClickListener {
            hideNavBtns(view)
        }
        view.btnStop.setOnClickListener {
            clickBtnStop(view)
        }
        view.btnForward.setOnClickListener {
            clickBtnForward(view)
        }
        view.btnBackward.setOnClickListener {
            clickBtnBackward(view)
        }
        view.btnFaster.setOnClickListener {
            clickBtnFaster(view)
        }
        view.btnSlower.setOnClickListener {
            clickBtnSlower(view)
        }
        view.btnVolumeOnOff.setOnClickListener {
            clickBtnMute(view)
        }
        return view
    }

    private fun clickBtnMute(view: View) {
        val am = this.context?.getSystemService(Context.AUDIO_SERVICE) as AudioManager
        val volumeLevel = am.getStreamVolume(AudioManager.STREAM_MUSIC)
        if (volumeLevel != 0) {
            am.setStreamMute(AudioManager.STREAM_MUSIC, true)
            view.btnVolumeOnOff.setImageResource(R.drawable.ic_baseline_volume_off)
            view.btnVolumeOnOff.tag = R.drawable.ic_baseline_volume_off
        } else {
            am.setStreamMute(AudioManager.STREAM_MUSIC, false)
            view.btnVolumeOnOff.setImageResource(R.drawable.ic_baseline_volume_up)
            view.btnVolumeOnOff.tag = R.drawable.ic_baseline_volume_up
        }
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnForward(view: View) {
        view.videoView.seekTo(view.videoView.duration)
        view.videoView.pause()
        view.btnPlay.setImageResource(R.drawable.ic_baseline_repeat)
        view.btnPlay.tag = R.drawable.ic_baseline_repeat
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnBackward(view: View) {
        view.videoView.pause()
        view.videoView.seekTo(0)
        view.videoView.tag = "pause"
        view.btnPlay.setImageResource(R.drawable.ic_baseline_play_arrow)
        view.btnPlay.tag = R.drawable.ic_baseline_play_arrow
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnFaster(view: View) {
        view.videoView.seekTo(view.videoView.currentPosition + 10000)
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnSlower(view: View) {
        view.videoView.seekTo(view.videoView.currentPosition - 10000)
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnStop(view: View) {
        view.videoView.pause()
        view.videoView.seekTo(0)
        view.videoView.tag = "first"
        view.btnPlay.setImageResource(R.drawable.ic_baseline_play_arrow)
        view.btnPlay.tag = R.drawable.ic_baseline_play_arrow
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun hideNavBtns(view: View) { // Скрыть-показать панель
        if (view.clNavBtns.visibility == View.VISIBLE) {
            view.clNavBtns.visibility = View.GONE
            Timer().schedule(MyTimerTask(), 5000)
        } else if (view.clNavBtns.visibility == View.GONE) {
            view.clNavBtns.visibility = View.VISIBLE
            Timer().schedule(MyTimerTask(), 5000)
        }
    }

    private fun clickBtnContinue(view: View) { // Продолжить после паузы
        view.videoView.start()
        view.btnPlay.setImageResource(R.drawable.ic_baseline_pause)
        view.btnPlay.tag = R.drawable.ic_baseline_pause
        view.videoView.tag = "play"
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnPause(view: View) { // Поставить паузу
        view.videoView.pause()
        view.btnPlay.setImageResource(R.drawable.ic_baseline_play_arrow)
        view.btnPlay.tag = R.drawable.ic_baseline_play_arrow
        view.videoView.tag = "pause"
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnRepeat(view: View) { // Начать заново
        view.videoView.resume()
        view.btnPlay.setImageResource(R.drawable.ic_baseline_pause)
        view.btnPlay.tag = R.drawable.ic_baseline_pause
        view.videoView.tag = "play"
        Timer().schedule(MyTimerTask(), 5000)
    }

    private fun clickBtnPlay(view: View, uri: Uri) { // Запустить проигрывание
        // Показываем окно загрузки
        val mDialog = ProgressDialog(activity)

        try {
            mDialog.setMessage("Пожалуйста, подождите...")
            mDialog.setCanceledOnTouchOutside(false)
            mDialog.show()
            view.videoView.setVideoURI(uri)
            view.videoView.setOnCompletionListener {
                view.btnPlay.setImageResource(R.drawable.ic_baseline_repeat)
                view.btnPlay.tag = R.drawable.ic_baseline_repeat
                if (view.clNavBtns.visibility == View.GONE) {
                    view.clNavBtns.visibility = View.VISIBLE
                }
            }
        } catch (e: Exception) {
            Snackbar.make(view, resources.getString(R.string.error), Snackbar.LENGTH_LONG)
                .show() // В случае ошибки показать Snackbar с текстом
        }
        view.videoView.requestFocus()
        // Проигрыватель готов к воспроизведению
        view.videoView.setOnPreparedListener { mediaPlayer ->
            mDialog.dismiss()
            mediaPlayer.isLooping = false
            if (mediaPlayer.videoHeight > mediaPlayer.videoWidth) {
                view.videoView.layoutParams =
                    ConstraintLayout.LayoutParams(
                        ConstraintLayout.LayoutParams.WRAP_CONTENT,
                        ConstraintLayout.LayoutParams.MATCH_PARENT
                    )
                val constraintSet = ConstraintSet()
                constraintSet.clone(view as ConstraintLayout)
                constraintSet.connect(view.videoView.id, ConstraintSet.LEFT, view.id, ConstraintSet.LEFT, 0)
                constraintSet.connect(view.videoView.id, ConstraintSet.RIGHT, view.id, ConstraintSet.RIGHT, 0)
                constraintSet.applyTo(view)
            } else {
                view.videoView.layoutParams =
                    ConstraintLayout.LayoutParams(
                        ConstraintLayout.LayoutParams.MATCH_PARENT,
                        ConstraintLayout.LayoutParams.WRAP_CONTENT
                    )
                val constraintSet = ConstraintSet()
                constraintSet.clone(view as ConstraintLayout)
                constraintSet.connect(view.videoView.id, ConstraintSet.TOP, view.id, ConstraintSet.TOP, 0)
                constraintSet.connect(view.videoView.id, ConstraintSet.BOTTOM, view.id, ConstraintSet.BOTTOM, 0)
                constraintSet.applyTo(view)
            }
            view.videoView.start()
            view.btnPlay.setImageResource(R.drawable.ic_baseline_pause)
            view.btnPlay.tag = R.drawable.ic_baseline_pause
            view.videoView.tag = "play"
            Timer().schedule(MyTimerTask(), 5000)
        }
    }

    internal inner class MyTimerTask : TimerTask() {

        override fun run() {
            activity!!.runOnUiThread { view!!.clNavBtns.visibility = View.GONE }
        }
    }

    companion object {
        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param param1 Parameter 1.
         * @param param2 Parameter 2.
         * @return A new instance of fragment PortraitFragment.
         */
        // TODO: Rename and change types and number of parameters
        @JvmStatic
        fun newInstance(param1: String, param2: String) =
            PortraitFragment().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putString(ARG_PARAM2, param2)
                }
            }
    }
}
